File: 1390_Four_Divisors/solution.c
int sum(int* arr, int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) sum += arr[i];
    return sum;
}

int get_divisors(int n) {
    int* divisors = (int*)malloc(4 * sizeof(int));
    if (divisors == NULL) return 0;
    
    int index = 0;

    int i = 1;
    while (i * i <= n) {
        if (n % i == 0) {
            if (index >= 4) {
                free(divisors);
                return 0;
            }
            divisors[index++] = i;
            
            if (i * i != n) {
                if (index >= 4) {
                    free(divisors);
                    return 0;
                }
                divisors[index++] = n / i;
            }
        }
        i += 1;
    }
    
    int result = (index == 4) ? sum(divisors, index) : 0;
    free(divisors);
    return result;
}

int sumFourDivisors(int* nums, int numsSize) {
    int result = 0;
    for (int i = 0; i < numsSize; i++) result += get_divisors(nums[i]);
    return result;
}

File: 1390_Four_Divisors/solution.py
class Solution:
    def sumFourDivisors(self, nums: List[int]) -> int:
        def get_sum(n):
            divisors = set()

            i = 1
            while i * i <= n:
                if n % i == 0:
                    divisors.add(i)
                    divisors.add(n // i)

                    if len(divisors) > 4: return 0
                i += 1
            return sum(divisors) if len(divisors) == 4 else 0

        result = 0
        for num in nums:
            result += get_sum(num)
        return result
